### PR TITLE
Add a CLI command to analyze execution times of classification hook

### PR DESCRIPTION
**Pull request by La0**
With this PR you can run `mozci push perf` to generate a CSV file with data about ALL classification tasks from a mozci hook.

Columns are :

-  branch
-  revision
-  push ID
-  task ID
-  task creation date
-  time taken by all runs in seconds

This allow a developer to analyze the execution time taken by classification for a given push (or range of pushes) over a period of time.

The command uses cache to avoid re-requesting completed tasks

### CODE REVIEW

**Comment by La0** [comment]
WDYT @marco-c & @ahal : I've now used it several times to build the CSV file to produce stats & analysis. Should we get this merged or just close the PR ?

**Comment by marco-c** [comment]
I think we should get it merged, it can be useful in the future too.

File: mozci/console/commands/push.py [hunk 1]
```diff
@@ -305,14 +314,160 @@ def handle(self):
...
+
+    def list_classification_tasks(self, group_id):
+        cache_key = f"perf/task_group/{group_id}"
+
+        # Check cache
+        tasks = config.cache.get(cache_key)
+        if tasks is None:
+            queue = taskcluster.Queue(get_taskcluster_options())
+            try:
+                tasks = queue.listTaskGroup(group_id).get("tasks", [])
```
Comment (marco-c) [inline mozci/console/commands/push.py:429]: Do we need pagination or is it automatically handled by the taskcluster library?

**Review by marco-c — APPROVED** [review]

File: mozci/console/commands/push.py (hunk 1 above)
Comment (La0) [reply mozci/console/commands/push.py:429]: Argh pagination is not handled automatically

Comment (La0) [reply mozci/console/commands/push.py]: Just added support for pagination

**Review by marco-c — APPROVED** [review]

**Pull request merged**